website/payments/models.py

```diff
@@ -92,6 +116,74 @@ def __str__(self):
         return _("Payment of {amount}").format(amount=self.amount)
 
 
+def _default_batch_description():
+    now = timezone.now()
+    return f"Thalia Pay payments for {now.year}-{now.month}"
+
+
+class Batch(models.Model):
+    """
+    Describes a batch of payments for export
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    processed = models.BooleanField(verbose_name=_("processing status"), default=False,)
+
+    processing_date = models.DateTimeField(
+        verbose_name=_("processing date"), blank=True, null=True,
+    )
+
+    description = models.TextField(
+        verbose_name=_("description of batch"), default=_default_batch_description,
+    )
+
+    def save(
+        self, force_insert=False, force_update=False, using=None, update_fields=None
+    ):
+        if self.processed and not self.processing_date:
+            self.processing_date = timezone.now()
+        super().save(force_insert, force_update, using, update_fields)
+
+    def get_absolute_url(self):
+        return reverse("admin:payments_batch_change", args=[str(self.pk)])
+
+    def start_date(self) -> datetime.datetime:
+        return self.payments_set.earliest("processing_date").processing_date
+
+    start_date.admin_order_field = "first payment in batch"
+    start_date.short_description = _("first payment in batch")
+
+    def end_date(self) -> datetime.datetime:
+        return self.payments_set.latest("processing_date").processing_date
+
+    end_date.admin_order_field = "last payment in batch"
+    end_date.short_description = _("last payment in batch")
+
+    def total_amount(self) -> Decimal:
+        return sum([payment.amount for payment in self.payments_set.all()])
+
+    total_amount.admin_order_field = "total amount"
+    total_amount.short_description = _("total amount")
+
+    def payments_count(self) -> Decimal:
+        return self.payments_set.all().count()
+
+    payments_count.admin_order_field = "payments count"
+    payments_count.short_description = _("payments count")
```

Comment:
These functions could use @property

Reply:
When they are a property, you can't give a short_description as far as I could find.